trl/trainer/alignprop_config.py

```diff
@@ -2,93 +2,119 @@
 import sys
 import warnings
 from dataclasses import dataclass, field
-from typing import Literal, Optional
+from typing import Any, Dict, Literal, Optional, Tuple
+
+from transformers import TrainingArguments
 
-from ..core import flatten_dict
 from ..import_utils import is_bitsandbytes_available, is_torchvision_available
 
 
 @dataclass
-class AlignPropConfig:
-    """
-    Configuration class for AlignPropTrainer
+class AlignPropConfig(TrainingArguments):
+    r"""
+    Configuration class for the [`AlignPropTrainer`].
+
+    Using [`~transformers.HfArgumentParser`] we can turn this class into
+    [argparse](https://docs.python.org/3/library/argparse#module-argparse) arguments that can be specified on the
+    command line.
+
+    Args:
+        exp_name (`str`, *optional*, defaults to `os.path.basename(sys.argv[0])[: -len(".py")]`):
```

Comment:
Note to self: I'm not sure why we need this, since we have `run_name` in `transformers.TrainingArguments`

Reply:
Agree. See https://github.com/huggingface/trl/pull/1944#discussion_r1736080674